zarrs/src/array/array_async_readable_writable.rs

```diff
@@ -278,11 +278,11 @@ impl<TStorage: ?Sized + AsyncReadableWritableStorageTraits + 'static> Array<TSto
 
             let store_chunk = |chunk_indices: Vec<u64>| {
                 let chunk_subset = self.chunk_subset(&chunk_indices).unwrap(); // FIXME: unwrap
-                let overlap = unsafe { array_subset.overlap_unchecked(&chunk_subset) };
+                let overlap = array_subset.overlap(&chunk_subset).unwrap(); // FIXME: unwrap
                 let chunk_subset_in_array_subset =
-                    unsafe { overlap.relative_to_unchecked(array_subset.start()) };
+                    overlap.relative_to(array_subset.start()).unwrap();
                 let array_subset_in_chunk_subset =
-                    unsafe { overlap.relative_to_unchecked(chunk_subset.start()) };
+                    overlap.relative_to(chunk_subset.start()).unwrap();
```

Comment:
More `unwrap` in place of `unchecked` didn't seem like a regression necessarily...could refactor so that all errors are handled here but seemed like an all-or-nothing thing and I wanted to keep this PR tight

Reply:
Don't worry about it in this PR, but I am gradually trying to hoover up the `unwrap()` in `zarrs` and replace them with `expect()` that clarifies why they should not panic in the same way that SAFETY docs clarify why `_unchecked` is okay to use.